core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -112,10 +112,10 @@ void beginParallelFor(const std::string& kernelPrefix, const uint32_t devID,
     Experimental::begin_context(context_id);
     Experimental::VariableValue contextValues[] = {
         Experimental::make_variable_value(
-            Experimental::kernel_name_context_variable_id,
-            kernelPrefix.c_str()),
+            Experimental::kernel_name_context_variable_id, kernelPrefix),
         Experimental::make_variable_value(
-            Experimental::kernel_type_context_variable_id, "parallel_for")};
+            Experimental::kernel_type_context_variable_id,
+            std::string("parallel_for"))};
```

Comment:
There is no need for this change

Reply:
Building the tutorial exercise I'm working on, if I remove that I get linker errors:

```
/home/dzpolia/src/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-7.1.0/binutils-2.34-tyakqk6dwdx2nd5ybnmfpayrszna6yj5/bin/ld: /home/dzpolia/src/kokkos-tutorials/Intro-Full/Exercises/tuning/Begin/libkokkos.a(Kokkos_Profiling.o): in function `Kokkos::Tools::beginParallelFor(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned int, unsigned long*)':
/ascldap/users/dzpolia/Kokkos/kokkos/core/src/impl/Kokkos_Profiling.cpp:113: undefined reference to `Kokkos::Tools::Experimental::make_variable_value(unsigned long, char const*)'
```

Reply:
https://github.com/kokkos/kokkos/blob/bd5eb149b6720c684df09bb6eacaa40b100c8051/core/src/impl/Kokkos_Profiling_Interface.hpp#L188